### PR TITLE
Manifest Kubernetes resources for `reset` command

### LINKED ISSUE #556: Manifest Kubernetes resources for `reset` command
Two modes are supported now:
### argo
One idea would be to manifest the Cleanup Jobs (Reset Jobs) with no PostDelete hook + the KafkaTopic resources.
This will initiate a delete auf the streams/producer apps and replace the previous PostDelete hooked Cleanup Jobs with the Reset Jobs (no hooks) and deploy them.

### manifest
Manifest reset jobs + the Kafka topics

### PR DESCRIPTION
**Pull request by raminqaf**
Closes #556

### CODE REVIEW

**Comment by disrupted** [comment]
please rename the PR

File: kpops/api/__init__.py [hunk 1]
```diff
@@ -120,6 +120,31 @@ def manifest_destroy(
...
 
+def manifest_reset(
+    pipeline_path: Path,
+    dotenv: list[Path] | None = None,
+    config: Path = Path(),
+    steps: set[str] | None = None,
+    filter_type: FilterType = FilterType.INCLUDE,
+    environment: str | None = None,
+    verbose: bool = True,
+    operation_mode: OperationMode = OperationMode.MANIFEST,
```
Comment (disrupted) [inline kpops/api/__init__.py:131]: what happens if the user would pass `OperationMode.MANAGED`? does that even make sense?

File: tests/pipeline/test_manifest.py [hunk 2]
```diff
@@ -231,6 +231,50 @@ def test_manifest_destroy_python_api(
...
+                str(RESOURCE_PATH / "manifest-pipeline" / PIPELINE_YAML),
+                "--operation-mode",
+                "argo",
+            ],
+            catch_exceptions=False,
+        )
+        assert result.exit_code == 0, result.stdout
+        snapshot.assert_match(result.stdout, MANIFEST_YAML)
+
+    def test_manifest_rest_python_api(self, capsys: CaptureFixture, snapshot: Snapshot):
```
Comment (disrupted) [inline tests/pipeline/test_manifest.py:262]:
```suggestion
    def test_manifest_reset_python_api(self, capsys: CaptureFixture, snapshot: Snapshot):
```

File: kpops/api/__init__.py (hunk 1 above)
Comment (raminqaf) [reply kpops/api/__init__.py:131]: Passing `OperationMode.MANIFEST` or `OperationMode.ARGO` would be the same. Passing the OperationMode MANAGED needs to make more sense for the manifest functions.

**Review by disrupted — APPROVED** [review]

**Pull request merged**